event-bus/cassandra/src/main/java/org/apache/james/events/CassandraEventDeadLetters.java

```diff
@@ -21,21 +21,38 @@
 
 import jakarta.inject.Inject;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
 
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class CassandraEventDeadLetters implements EventDeadLetters {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraEventDeadLetters.class);
 
     private final CassandraEventDeadLettersDAO cassandraEventDeadLettersDAO;
     private final CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO;
+    private final EventSerializer eventSerializer;
+    private final EventDeserializers deserializer;
+
+    public CassandraEventDeadLetters(CassandraEventDeadLettersDAO cassandraEventDeadLettersDAO,
+                                     CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO,
+                                     EventSerializer eventSerializer) {
+        this(cassandraEventDeadLettersDAO, cassandraEventDeadLettersGroupDAO, eventSerializer, new EventDeserializers(ImmutableSet.of(eventSerializer)));
+    }
 
     @Inject
     public CassandraEventDeadLetters(CassandraEventDeadLettersDAO cassandraEventDeadLettersDAO,
-                                     CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO) {
+                                     CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO,
+                                     EventSerializer eventSerializer,
+                                     EventDeserializers deserializer) {
```

Comment:
I still don't understand why we need separate for serialization and deserialization, and what the difference is between them.

Reply:
How about support for multiple `EventSerializer` as well? I found weird as we support multiple `EventDeserializer` (s) but not for the serializer.

Maybe just create an `EventSerializersAggregate` or something to support multiple serializers + deserializers?